app/javascript/packs/containers/ReviewModeContainer.jsx

```diff
@@ -7,10 +7,89 @@ import React from 'react';
 
 import QuestionContainer from './QuestionContainer';
 
+import { getAnswerForQuestionAndBuilding } from '../selectors/answersSelector';
+import { getDependentQuestionsForOption } from "../selectors/questionsSelector";
 import { connect } from 'react-redux';
 import { getQuestionsByBuilding } from '../selectors/questionsSelector';
 
+import { post, patch } from '../fetch/requester';
+
+async function postDelegations(delegations, callback) {
+  var body = {
+    "delegations": delegations,
+  };
+
+  try {
+    let resp = await post('/api/delegations', body);
+    callback(true);
+  } catch (error) {
+    callback(false);
+  }
+}
+
 class ReviewModeContainer extends React.Component {
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      status_string: "Delegations have not been saved yet.",
+    }
+  }
+
+  // recursively find all dependent questions, return a list
+  getDependentQuestion(parentQuestion) {
+    var potentialChildren = [].concat(
+        ...Object.values(parentQuestion.options).map((option) => {
+          return this.props.getDependentQuestion(option.id, parentQuestion.question_type);
+        })
+    );
+
+    if (potentialChildren.length == 0) {
+      return [];
+    } else {
+      return potentialChildren.concat(potentialChildren.map(this.getDependentQuestion));
```

Comment:
`this` is undefined when `potentialChildren.map` calls this callback. Pass an arrow function into `map` instead to bind implicitly bind this function!